src/main/webapp/script.js

```diff
@@ -1,88 +1,185 @@
   }
-
-  searchInput.on('input', function() {
-    sessionStorage.setItem('searchVal', $(this).val());
-    getDashboardItems();
-  });
 }
 
 /**
- * Loads the stored values for the sort type of the sort selector and adds a
- * callback to load the entertainment items when the selector changes value.
+ * Loads the stored values for the sort type of the sort selector from
+ * sessionStorage
  */
-function loadSortValue() {
-  const sortSelector = $('#sortType');
+function initializeSortSelector() {
   const sortVal = sessionStorage.getItem('sortVal');
 
   if (sortVal !== null) {
-    sortSelector.val(sortVal);
+    $('#sortType').val(sortVal);
   }
+}
 
-  sortSelector.change(function() {
+/**
+ * Fetches for the entertainment items from DashboardServlet and the favorite
+ * item ids from FavoriteItemServlet to initialize the dashboard with items that
+ * have correct like button state.
+ */
+async function initializeDashboard() {
+  Promise
+      .allSettled([
+        fetch(
+            '/dashboard?searchValue=' + $('#searchValue').val() +
+            '&sortType=' + $('#sortType').val()),
+        fetch('/favorite-item')
+      ])
+      .then(async (results) => {
+        dashboardResult = results[0];
+        favResult = results[1];
+
+        if (!dashboardResult.value.ok) {
+          console.log(
+              'Failed to fetch Entertainment Items from DashboardServlet: ' +
+              dashboardResult.reason);
+          return;
+        }
+      
+        try {
+          const entertainmentItems = await dashboardResult.value.json();
+          const favoriteItemIds =
+              favResult.value.ok ? await favResult.value.json() : [];
+
+          setupSeachInputCallback(favoriteItemIds);
+          setupSortSelectorCallback(favoriteItemIds);
+          updateDashboardItems(entertainmentItems, favoriteItemIds);
+        } catch (error) {
+          console.log('Failed to populate dashboard: ' + error);
+        }
+      });
+}
+
+/**
+ * Adds an input value callback to the search input for it to update its value
+ * to sessionStorage and fetch for new items.
+ *
+ * @param { Array } favoriteItemIds - the list of entertainment item Ids that
+ *     have been liked by the logged in user
+ */
+function setupSeachInputCallback(favoriteItemIds) {
+  $('#searchValue').on('input', function() {
+    sessionStorage.setItem('searchVal', $(this).val());
+    getEntertainmentItems(favoriteItemIds);
+  });
+}
+
+/**
+ * Adds a change value callback to the sort selector for it to update its value
+ * to sessionStorage and fetch for new items.
+ *
+ * @param { Array } favoriteItemIds - the list of entertainment item Ids that
+ *     have been liked by the logged in user
+ */
+function setupSortSelectorCallback(favoriteItemIds) {
+  $('#sortType').change(function() {
     sessionStorage.setItem('sortVal', $(this).val());
-    getDashboardItems();
+    getEntertainmentItems(favoriteItemIds);
   });
 }
 
 /**
```

Comment:
Overall I think the code is fine, but I'll share just a thought.

What if you create a global variable for the array favoriteItemIds?
That way, you don't have to wait for the promises inside `initializeDashboard` to complete before calling these setup methods. These setup methods can be merged with the initialize methods above.

Reply:
I think I am going to keep passing the parameter for now instead of the global variable. Might reconsider in future PR if it seems worth it.